rmsutil/system.py

```diff
@@ -0,0 +1,31 @@
+import contextlib
+import copy
+import os
+import sys
+
+
+@contextlib.contextmanager
+def push_env(**kwargs):
+    try:
+        old_env = os.environ.copy()
+
+        for key, value in kwargs.items():
+            os.environ[key] = value
+
+        yield
+
+    finally:
+        os.environ.clear()
+        os.environ.update(old_env)
+
+
+@contextlib.contextmanager
+def push_argv(*args):
+    old_argv = copy.copy(sys.argv)
```

Comment:
Ignorable nitpick: since you're slicing sys.argv, you're getting a new list, so I don't think you need a copy.